core/src/main/scala/dagr/core/config/Configuration.scala

```diff
@@ -40,7 +40,9 @@ import scala.reflect.runtime.universe.{TypeTag, typeOf}
   * Companion object to the Configuration trait that keeps track of all configuration keys
   * that have been requested so that they can be reported later if desired.
   */
-private[core] object Configuration extends ConfigurationLike {
+object Configuration extends ConfigurationLike {
+  // Developer Note: [[Configuration]] is not private so that [[Configuration.Keys]] is public
```

Comment:
I'm not convinced of this - the point of having this private was to prevent pipeline developers from using it instead of mixing in `Configuration`.  Can we leave it `private[core]` or perhaps even `private[dagr]`?